app/lib/account_deletion_config.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module AccountDeletionConfig
+  module_function
+
+  def config
+    @config ||= load_config
+  end
+
+  def valid?
+    @valid ||= check_validity
+  end
+
+  def load_config
+    config = ThreeScale.config.ttl.account_deletion.each_with_object({}) do |(key, value), collection|
+      collection[key.to_sym] = value if value.is_a?(Integer)
+    end
+    ActiveSupport::OrderedOptions.new.merge(config)
+  end
+
+  def check_validity
+    valid = %i[account_suspension account_inactivity contract_unpaid_time].all? { |key| config.key?(key) }
+    inform_invalid_configuration if !valid && ThreeScale.config.ttl.account_deletion.present?
```

Comment:
Do we need a public method for the warning, even a method at all?

Reply:
I can not make it private because it is a Module with `module_function`, but I can remove it 👍